frame/support/procedural/src/expand_after.rs

```diff
@@ -0,0 +1,131 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of macro `expand_after`.
+
+use proc_macro2::{Group, Span, TokenStream, TokenTree};
+use std::iter::once;
+use syn::spanned::Spanned;
+
+pub fn expand_after(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	let def = syn::parse_macro_input!(input as ExpandAfterDef);
+	let expand_in_span = def.expand_in.span();
+
+	match expand_in_stream(&def.expand_after, &mut Some(def.expand_with), def.expand_in) {
+		Ok(stream) => stream.into(),
+		Err(_) => {
+			let msg = format!("Cannot find pattern `{:?}` in given token stream", def.expand_after);
+			syn::Error::new(expand_in_span, msg).to_compile_error().into()
```

Comment:
It may make more sense to emit this error one level down, i.e. inside of `expand_in_stream` rather than here in `expand_after`, since it's more localized to the place where the error occurred. The span can just be retrieved from `stream.span()`.

Reply:
yes done in https://github.com/paritytech/substrate/pull/9681/commits/89cf84118b5ebf3a59bf0dc9001a20357efe822c

Reply:
hmm thinking again I was probably thinking it is better not to build the error for each call to expand_in_stream, and only build it an the end.